src/components/validation/validators/component_spec/sources.rs

```diff
@@ -226,3 +229,16 @@ fn validate_component_sent_event_bytes_total(
         expected_bytes,
     )
 }
+
+fn validate_component_discarded_events_total(
+    telemetry_events: &[Event],
+    runner_metrics: &RunnerMetrics,
+) -> Result<Vec<String>, Vec<String>> {
+    let expected_dropped = runner_metrics.discarded_events_total;
```

Comment:
Note- the expected value is not currently being set.

This is a rather tricky one to validate since it requires forcing the stream to close, after we sent the events to the source and before it sent them to the downstream.